simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+
+    parser.add_argument(
+        '--people',
+        default = DEFAULT_PEOPLE,
+        type = checkNonNegativeInt,
+        help = 'Number of people in your dataset.')
+
+    parser.add_argument(
+        '--places',
+        default = DEFAULT_PLACES,
+        type = checkNonNegativeInt,
+        help = 'Number of places in your dataset.')
+
+    parser.add_argument(
+        '--placesLivedMean',
+        default = DEFAULT_PLACES_LIVED_MEAN,
+        type = checkNonNegativeInt,
+        help = 'Mean of the number of places a person has lived in.')
+
+    parser.add_argument(
+        '--placesLivedSD',
+        default = DEFAULT_PLACES_LIVED_SD,
+        type = checkNonNegativeFloat,
+        help = 'Standard deviation of the number of places a person has lived in.')
+
+    parser.add_argument(
+        '--randomSeed',
+        default = np.random.randint(DEFAULT_MAX_RAND_INT),
+        type = int,
+        help = 'Seed for the random number generator used in the script.')
+
+    parser.add_argument(
+        '--targetSplit',
+        default = DEFAULT_TARGET_SPLIT,
+        type = checkValidSplit,
+        help = 'Fraction of the truth data to be used as targets.')
+
+    args = parser.parse_args()
+    return args
+
+
+def setRandomSeed(seed):
+    """
+    Set the seed for all random number generators in the script.
+    """
+    np.random.seed(seed)
+
+
+def getTruncatedNormal(mean, sd, low, upp):
+    """
+    Return truncated normal distribution.
+    """
+    return truncnorm(
+        (low - mean) / sd, (upp - mean) / sd, loc = mean, scale = sd)
+
+
+class Person():
+    """
+    Person class for the dataset.
+    """
+
+    def __init__(self):
+        self.index = None
+        self.globalLikes = None
+        self.lived = None
+        self.localLikes = None
+
+
+class DataGen():
+    """
+    Data generator class for simple-acquaintances example.
+    """
+
+    def __init__(
+            self,
+            numberOfPeople,
+            numberOfPlaces,
+            numberOfGlobal,
+            numberOfLocal,
+            placesLivedMean,
+            placesLivedSD,
+            localLikesVariance,
+            targetSplit,
+            randomSeed,
+            outputDir):
+        self.numberOfPeople = numberOfPeople
+        self.places = [i for i in range(numberOfPlaces)]
+        self.globalThings = [i for i in range(numberOfGlobal)]
+        self.localThings = [i for i in range(numberOfLocal)]
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = numberOfPlaces
+        self.localLikesVariance = localLikesVariance
+        self.targetSplit = targetSplit
+        self.randomSeed = randomSeed
+        self.outputDir = outputDir
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # Fill in placesAffectingLocalThings.
+
+        setRandomSeed(self.randomSeed)
```

Comment:
Instead of seeding NumPy's global RNG and hoping people don't mess with it, just make your own instance of an RNG:
https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.random.RandomState.html